op-node/rollup/derive/kromampt_upgrade_transactions_test.go

```diff
@@ -0,0 +1,175 @@
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+func TestMPTSourcesMatchSpec(t *testing.T) {
+	for _, test := range []struct {
+		source       UpgradeDepositSource
+		expectedHash string
+	}{
+		{
+			source:       deployBaseFeeVaultSource,
+			expectedHash: "0xdd882729c9833d55eb4ad39f08f578c0b92e171234e1b4ddc8f91d7878cd8686",
+		},
+		{
+			source:       updateBaseFeeVaultProxySource,
+			expectedHash: "0x747e2dc0ea5ed790a8746601b59c7588bfdce93e5d4dc0f6fd92a4d32c07596d",
+		},
+		{
+			source:       deployL1FeeVaultSource,
+			expectedHash: "0xb88c42fa962a1aa3d495761f85b93fd496a9607b57ffda85a1e3840b0d854794",
+		},
+		{
+			source:       updateL1FeeVaultProxySource,
+			expectedHash: "0xd4b39da88babd8b9c3d96bfc88b2c4be2a977c3b2629d0f2498fc290e1c523be",
+		},
+		{
+			source:       deploySequencerFeeVaultSource,
+			expectedHash: "0x4618df4b1692fcaee1da6b0532641b361f5dbfc3bb5714ae22802f989dc59c6f",
+		},
+		{
+			source:       updateSequencerFeeVaultProxySource,
+			expectedHash: "0xaf7e0d5c4b92ad5918fe238414031d8a82b87841ed70f1ba247d50121f9c04eb",
+		},
+		{
+			source:       deployL1BlockMPTSource,
+			expectedHash: "0x3c459db0fc6553f1b48a3b7f1e20de2e80d7f28976085817070699d1502dbe74",
+		},
+		{
+			source:       updateL1BlockMPTProxySource,
+			expectedHash: "0xb3a45c33c31a2322c4c4db99788dcae68606b98e4c90ea9fb3d1bc93ddfbf3bf",
+		},
+	} {
+		require.Equal(t, common.HexToHash(test.expectedHash), test.source.SourceHash())
+	}
+}
+
+func TestMPTNetworkTransactions(t *testing.T) {
+	// test chainId-specific configurations
+	tests := []struct {
+		name    string
+		chainId *big.Int
+	}{
+		{
+			name:    "Local devnet configuration",
+			chainId: big.NewInt(KromaLocalDevnetChainID),
+		},
+		{
+			name:    "Kroma configuration",
+			chainId: big.NewInt(params.KromaMainnetChainID),
+		},
+		{
+			name:    "Kroma Sepolia configuration",
+			chainId: big.NewInt(params.KromaSepoliaChainID),
+		},
+		{
+			name:    "Kroma Holesky configuration",
+			chainId: big.NewInt(params.KromaDevnetChainID),
+		},
+		{
+			name:    "Unsupported chain. Defaults to local devnet",
+			chainId: big.NewInt(9999),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			upgradeTxns, err := KromaMPTNetworkUpgradeTransactions(tt.chainId)
+			require.NoError(t, err)
+			deploymentData, err := getDeploymentData(tt.chainId)
+			require.NoError(t, err)
+			require.Len(t, upgradeTxns, 8)
+
+			deployBaseFeeVaultSender, deployBaseFeeVault := toDepositTxn(t, upgradeTxns[1])
+			require.Equal(t, deployBaseFeeVaultSender, BaseFeeVaultDeployerAddress)
+			require.Equal(t, deployBaseFeeVaultSource.SourceHash(), deployBaseFeeVault.SourceHash())
+			require.Nil(t, deployBaseFeeVault.To())
+			require.Equal(t, uint64(1_000_000), deployBaseFeeVault.Gas())
+			require.Equal(t, fmt.Sprintf("0x%x", deploymentData.baseFeeVaultDeploymentBytecodeWithArg), hexutil.Bytes(deployBaseFeeVault.Data()).String())
```

Comment:
Same applys to below `fmt.Sprintf("0x%x", ~)`
```suggestion
			require.Equal(t, hexutil.Bytes(deploymentData.baseFeeVaultDeploymentBytecodeWithArg).String(), hexutil.Bytes(deployBaseFeeVault.Data()).String())
```